pykafka/producer.py

```diff
@@ -140,6 +141,14 @@ def __init__(self,
         self._update_lock = self._cluster.handler.Lock()
         self.start()
 
+        def cleanup(obj):
+            if obj._running:
+                obj.stop()
```

Comment:
@emmett9001 I had missed this part, so it seems that ``self.__exit__`` behaves the same way when we instantiate the class using ``with``, but it would require a refactor of our client code to use that, we use the same class instance in different parts of the code.